extensions/wikia/AdEngine/js/video/ooyalaAdSetProvider.js

```diff
@@ -0,0 +1,69 @@
+/*global define*/
+define('ext.wikia.adEngine.video.ooyalaAdSetProvider', [
+	require.optional('ext.wikia.adEngine.adContext'),
+	require.optional('ext.wikia.adEngine.video.vastUrlBuilder')
+], function (adContext, vastUrlBuilder) {
+	'use strict';
+
+	var FEATURED_VIDEO_SIZE = 640 / 480;
+
+	function generateSet(vpos, type, position, rv, correlator) {
+		return {
+			position_type: type,
+			position: position,
+			tag_url: vastUrlBuilder.build(FEATURED_VIDEO_SIZE, {
+				pos: 'FEATURED',
+				src: 'premium',
+				rv: rv
+			}, {
+				vpos: vpos,
+				correlator: correlator
+			})
+		};
+	}
+
+	function calculateRV(videoDepth, capping) {
+		return videoDepth === 1 ? 1 : Math.floor((videoDepth - 1) / capping) + 1;
+	}
+
+	function shouldPlayNextVideoAd(videoDepth, capping) {
+		return (videoDepth - 1) % capping === 0;
+	}
+
+	function isAbleToDisplayAds() {
+		return adContext && adContext.getContext() && adContext.getContext().opts.showAds && vastUrlBuilder;
+	}
+
+	function get(videoDepth, correlator) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)